go-controller/pkg/ovn/ovn.go

```diff
@@ -453,23 +453,25 @@ func (oc *Controller) iterateRetryPods(updateAll bool) {
 		if podEntry.ignore {
 			continue
 		}
+
 		pod := podEntry.pod
-		if !util.PodScheduled(pod) {
+		podDesc := fmt.Sprintf("[%s/%s/%s]", pod.UID, pod.Namespace, pod.Name)
+		// it could be that the Pod got deleted, but Pod's DeleteFunc has not been called yet, so don't retry
+		kPod, err := oc.watchFactory.GetPod(pod.Namespace, pod.Name)
+		if err != nil {
+			if e, ok := err.(*errors.StatusError); ok && e.ErrStatus.Code == http.StatusNotFound {
+				klog.Infof("%s pod not found in the informers cache, not going to retry pod setup", podDesc)
+				delete(oc.retryPods, uid)
+				continue
+			}
+		}
+
+		if !util.PodScheduled(kPod) {
```

Comment:
we check pod scheduled here and in ensurePod. We pass ensurePod the pod from the retry cache, so it should also fail the scheduled check there too. I think I need to push another revision to this PR